js/index.js

```diff
@@ -1,42 +1,76 @@
 // ITERATION 1
 
-function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
-
+function updateSubtotal(products) {
+  // console.log('Calculating subtotal, yey!');
   //... your code goes here
+  let totalArr = [];
+  for (let item of products) {
+    let price = item.querySelector('.price span');
+    let quantity = item.querySelector('.quantity input');
+    let subtotal = item.querySelector('.subtotal span');
+    price = parseFloat(price.innerHTML);
+    quantity = parseFloat(quantity.value);
+    subtotal.innerHTML = `${price * quantity}`;
+    totalArr.push(price * quantity);
+  }
+  return totalArr.reduce((total, item) => total + item, 0);
 }
 
 function calculateAll() {
+  const totalTag = document.querySelector('#total-value span');
   // code in the following two lines is added just for testing purposes.
   // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
+  // ITERATION 1 & 2
 
-  // ITERATION 2
-  //... your code goes here
+  const products = document.getElementsByClassName('product');
 
   // ITERATION 3
-  //... your code goes here
+  let total = updateSubtotal(products);
+  totalTag.innerHTML = total;
```

Comment:
Aqui estaria mejor usar textContent por que su valor no se analiza como HTML 
```suggestion
  totalTag.textContent = total;
```